core/tests/unit/kmnkbd/debug_api.cpp

```diff
@@ -423,6 +424,7 @@ void test_backspace_markers() {
     {KM_CORE_CT_END}
   };
   try_status(km_core_context_set(km_core_state_context(test_state), marker_context));
+  try_status(km_core_context_set(km_core_state_app_context(test_state), marker_context)); // TODO: this needs to remove markers
```

Comment:
if only there were a function for that.

`remove_markers(std::u32string &str /*... */)` could be used, leaving all other params default.

Reply:
apart from `remove_markers()` working on the internal string representation and not an `km_core_context_item` array.

Reply:
TODO: remove markers here and elsewhere where app_context is set directly in unit tests. Consider adding assertions to verify that app_context never contains markers. Or consider making the setter strip markers automatically.

Reply:
See #10388.